### PR TITLE
logrotate every hour

### PR DESCRIPTION
**Pull request by emord**
I think this should help solve the space issues we've seen recently on proxy workers. 

By default logrotate is only run every day. This moves it to every hour. This means that if the log is at 749M when it gets run, then it could balloon over the next night and not be caught by th erotation

I couldn't make this change in the normal ansible-logrotate because that's a submodule to a repo we don't control. that should get moved to galaxy at some point.

@javierwilson @snopoke @sravfeyn

### CODE REVIEW

File: ansible/deploy_proxy.yml [hunk 1]
```diff
@@ -90,6 +90,8 @@
             - missingok
             - compress
             - delaycompress
-            - copytruncate
             - nocreate
             - notifempty
+            - sharedscripts
```
Comment (emord) [inline ansible/deploy_proxy.yml]: This means the post rotate script is only run once

File: ansible/deploy_proxy.yml [hunk 2]
```diff
@@ -90,6 +90,8 @@
...
+          scripts:
+            postrotate: "[ -s /run/nginx.pid ] && kill -USR1 `cat /run/nginx.pid`"
```
Comment (emord) [inline ansible/deploy_proxy.yml]: Tells nginx to change its file handles for logging

File: ansible/deploy_proxy.yml [hunk 3]
```diff
@@ -90,6 +90,8 @@
             - missingok
             - compress
             - delaycompress
-            - copytruncate
```
Comment (emord) [inline ansible/deploy_proxy.yml]: having this meant that htere was a portion of time that the we lost logs...

**Comment by emord** [comment]
deployed to nic and ctrls

File: ansible/roles/logrotate/tests/test.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+- hosts: localhost
+  remote_user: root
+  roles:
+    - logrotate
```
Comment (snopoke) [inline ansible/roles/logrotate/tests/test.yml]: Nice, will this actually run?

Comment (emord) [reply ansible/roles/logrotate/tests/test.yml]: Good question. I'm not sure. I just  used `ansible-galaxy init logrotate` to initialize the role

**Comment by emord** [comment]
thanks. added it to deploy_db

**Pull request merged**